.github/workflows/test-head.yaml

```diff
@@ -12,6 +12,7 @@ jobs:
       JRUBY_OPTS: --dev
     steps:
       - uses: actions/checkout@v2
+      - run: sed -i~ -e '/spec\.required_ruby_version/d' ddtrace.gemspec
```

Comment:
Minor:

* Would it make sense to add a small comment saying something like "Skip gem maximum version restriction when testing with ruby-head", just to add a bit of context to this magic incantation?
* My sed-foo is not amazing; I couldn't figure out what the `-i~` does and how differs from `-i`

Reply:
`man sed` ;)

`-i` means "edit in place", and takes a value to append to the filename to backup the original. On GNU systems `-i` has this value optional, which means discard the original, but on BSD systems it's not, so the value would be `-e`. Coincidentally, `-e` is optional on both when there's only one pattern, so on BSD systems `sed -i -e 'pattern' file` works but silently breaks the user expectation by saving the original as `file-e`! So, that's a portable way to write inline replacements with sed. As for appending `~`, it's a historical convention not entirely unlike `.bak`.

Reply:
> "Skip gem maximum version restriction when testing with ruby-head"

Thought about it but given that it's in the `test-head.yaml file` (and on the other one there's a `if:` clause testing for `head`) I don't find this comment to be adding value (it sounds like a "what")

That said, mentioning the "why" context (being unable to include prereleases yet exclude releases) sounds like a good idea.

Reply:
Thanks for explaining the sed magic!

Also agree with your note on the comment -- I was thinking from the POV that we sometimes copy pasta CI stuff and add more versions and whatnot so the context of "this sed only makes sense for head" may be lost, but the better fix is definitely your suggestion of explaining why :)